data/dataset.py

```diff
@@ -116,6 +116,8 @@ def load_cache_dataset(
     dataset_cfg: DictConfig,
     dataset_name: str,
     split: str,
+    src_encoder: MidiEncoder,
+    tgt_encoder: MidiEncoder,
     force_build: bool = False,
 ) -> MyTokenizedMidiDataset:
```

Comment:
what do you think of this way of preparing a dataset? (By passing src and tgt encoders to load_cache_dataset function)

Reply:
The dataset creation logic now looks like this:
```py
    src_encoder = QuantizedMidiEncoder(quantization_cfg=cfg.dataset.quantization)
    tgt_encoder = DstartEncoder(bins=cfg.dstart_bins)
    train_translation_dataset = load_cache_dataset(
        dataset_cfg=cfg.dataset,
        dataset_name=cfg.dataset_name,
        split="train",
    )
    val_translation_dataset = load_cache_dataset(
        dataset_cfg=cfg.dataset,
        dataset_name=cfg.dataset_name,
        split="validation",
    )

    train_dataset = MyTokenizedMidiDataset(
        dataset=train_translation_dataset,
        dataset_cfg=cfg.dataset,
        src_encoder=src_encoder,
        tgt_encoder=tgt_encoder
    )
    val_dataset = MyTokenizedMidiDataset(
        dataset=val_translation_dataset,
        dataset_cfg=cfg.dataset,
        src_encoder=src_encoder,
        tgt_encoder=tgt_encoder
    )

```
I think it's pretty good.